### DISCUSSION TITLE
problem at legend function in Chart.js

### DISCUSSION THREAD

**Discussion by naoyoshi** [nuxt/nuxt#17377] [Questions]
Hi.
Though I don't know there's any person like me, I have a question.

Things which I want to do:
1) I use Chart.js 3.8
https://www.chartjs.org/docs/latest/configuration/legend.html

2) I create chart area inside Component as like below: 
import { thisExpression } from '@babel/types';
import Chart from 'chart.js/auto';

3) Every time, when I click one element in legend area, I want to fire to fetch data from server.
I want to achieve this fetching process by calling defined method.

Problem:
1) Inside Chart.plugins.legend.onCLick function, it can't call defined method in Component. 
It shows error as 'this.medhodName is not a function'.

2) Inside Chart.overrides[type].plugins.legend.onCLick function, it can't call this.$emit.

With Chart.js 2.8.x and simple JavaScript, I could call defined fetching method in Chart.plugins.legend.onCLick function directly.
I want to let Vue.js 3/Nuxt.js 3 method or vaiable detect to click legend of chart.

Any ideas?

**Comment by Flanker72** [comment]
If you can provide some sample code - that would be easier to solve the problem.
For now it seems that you are trying to use `this` to access the component, however the Composition API don't use `this` approach.
You can define your function in the root level of the component, define `emit` function and use them from the component. This may be implemented like this:
```
<script setup lang="ts">
const emit = defineEmits<{
  (event: "click"): void;
}>();

function myMethod() {
  console.log("My method");
}

onMounted(() => {
  Chart.overrides[type].plugins.legend.onClick = () => {
    myMethod();
    emit("click");
  }
});
</script>
```

**Reply by naoyoshi** [reply]
Thank you for reply.
